src/controller/AutoCommissioner.cpp

```diff
@@ -760,6 +760,7 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
                 }
             }
 
+            mNeedIcdRegistration = false;
```

Comment:
This is probably OK, but conceptually it would make more sense to either store this in mParams or reset it in `AutoCommissioner::SetCommissioningParameters` like we do other things we get from the device (e.g. the failsafe timer above).

Reply:
yes, it make sense, move it inside AutoCommissioner::SetCommissioningParameters